mypaint-brush.c

```diff
@@ -646,6 +646,29 @@ smallest_angular_difference(float a, float b)
 
     float base_radius = expf(mypaint_mapping_get_base_value(self->settings[MYPAINT_BRUSH_SETTING_RADIUS_LOGARITHMIC]));
 
+	if (self->settings_value[MYPAINT_BRUSH_SETTING_OFFSET_X]) {
+		x += self->settings_value[MYPAINT_BRUSH_SETTING_OFFSET_X] * base_radius;	
+	}
+	
+	if (self->settings_value[MYPAINT_BRUSH_SETTING_OFFSET_Y]) {
+		y += self->settings_value[MYPAINT_BRUSH_SETTING_OFFSET_Y] * base_radius;	
+	}
+	
+	if (self->settings_value[MYPAINT_BRUSH_SETTING_OFFSET_ANGLE]) {
+		x += base_radius * self->settings_value[MYPAINT_BRUSH_SETTING_OFFSET_ANGLE] * self->states[MYPAINT_BRUSH_STATE_DIRECTION_DY];
+		y += base_radius * self->settings_value[MYPAINT_BRUSH_SETTING_OFFSET_ANGLE] * -self->states[MYPAINT_BRUSH_STATE_DIRECTION_DX];	
+	}
+	
+	if (self->settings_value[MYPAINT_BRUSH_SETTING_OFFSET_ANGLE_2]) {
+		if (self->settings_value[MYPAINT_BRUSH_SETTING_OFFSET_ANGLE_2] < 0) {
+			self->settings_value[MYPAINT_BRUSH_SETTING_OFFSET_ANGLE_2] = 0;	
+		}
+		static int sign = +1;
```

Comment:
@AnTi-ArT I made a pull request to your branch, I think... :-).  I think my pull is rebasing your branch too or something.